#Indicadores de enfermedades crónicas en US"
>Nombre del Dataset: Dataset3_U.S._Chronic_Disease_Indicators.csv

##INTRODUCCIÓN
Se realiza una exploración de datos sobre los indicadores de enfermedades crónicas en US. El objetivo es crear un resumen que investigue las enfermedades crónicas más comunes en los EE. UU., señalando correlaciones, patrones y tendencia entre indicadores, localidad, Gastos hospitalarios, género, raza, etc. 


##PREGUNTAS A RESPONDER:
*   Analizar las Tendencias del consumos alcohol en la población y determinar si esta ha mejorado en el tiempo.
*   Cómo se correlacionan la insuficiencia renal con otros indicadores 
*   ¿Los estados que mostraron un aumento en el consumo de alcohol también mostraron un aumento en la insuficiencia renal?

In [1]:
import pandas as pd
import numpy as np

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
from google.colab import drive
import os
drive.mount('/content/gdrive') #montar la carpeta

Mounted at /content/gdrive


#Carga del dataset en formtato .csv 


In [4]:
%cd '/content/gdrive/MyDrive/ProyectoDataScience/'
df_uscronic= pd.read_csv('Dataset3_U.S._Chronic_Disease_Indicators.csv',sep=',')

/content/gdrive/.shortcut-targets-by-id/1JBjZRPlZTm-Ax9WZx4quXsWlJqPbkKqm/ProyectoDataScience


/usr/local/lib/python3.8/dist-packages/IPython/core/interactiveshell.py:3326: DtypeWarning: Columns (7,10,18,19,20,21,23,30,31,32,33) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


#Entendiendo la data
##Mostar el Dataset Preliminar (Sin transformaciones)



In [5]:
df_uscronic.head()

YearStart  YearEnd LocationAbbr LocationDesc DataSource    Topic  \
0       2015     2015           AK       Alaska      YRBSS  Alcohol   
1       2015     2015           AL      Alabama      YRBSS  Alcohol   
2       2015     2015           AR     Arkansas      YRBSS  Alcohol   
3       2015     2015           AZ      Arizona      YRBSS  Alcohol   
4       2015     2015           CA   California      YRBSS  Alcohol   

                  Question Response DataValueUnit     DataValueType  ...  \
0  Alcohol use among youth      NaN             %  Crude Prevalence  ...   
1  Alcohol use among youth      NaN             %  Crude Prevalence  ...   
2  Alcohol use among youth      NaN             %  Crude Prevalence  ...   
3  Alcohol use among youth      NaN             %  Crude Prevalence  ...   
4  Alcohol use among youth      NaN             %  Crude Prevalence  ...   

  LocationID  TopicID QuestionID DataValueTypeID  StratificationCategoryID1  \
0          2      ALC     ALC1_1         CrdPrev                    OVERALL   
1          1      ALC     ALC1_1         CrdPrev                    OVERALL   
2          5      ALC     ALC1_1         CrdPrev                    OVERALL   
3          4      ALC     ALC1_1         CrdPrev                    OVERALL   
4          6      ALC     ALC1_1         CrdPrev                    OVERALL   

   StratificationID1 StratificationCategoryID2 StratificationID2  \
0                OVR                       NaN               NaN   
1                OVR                       NaN               NaN   
2                OVR                       NaN               NaN   
3                OVR                       NaN               NaN   
4                OVR                       NaN               NaN   

  StratificationCategoryID3 StratificationID3  
0                       NaN               NaN  
1                       NaN               NaN  
2                       NaN               NaN  
3                       NaN               NaN  
4                       NaN               NaN  

[5 rows x 34 columns]

In [6]:
df_uscronic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 403984 entries, 0 to 403983
Data columns (total 34 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   YearStart                  403984 non-null  int64  
 1   YearEnd                    403984 non-null  int64  
 2   LocationAbbr               403984 non-null  object 
 3   LocationDesc               403984 non-null  object 
 4   DataSource                 403984 non-null  object 
 5   Topic                      403984 non-null  object 
 6   Question                   403984 non-null  object 
 7   Response                   79323 non-null   object 
 8   DataValueUnit              374119 non-null  object 
 9   DataValueType              403984 non-null  object 
 10  DataValue                  297817 non-null  object 
 11  DataValueAlt               273666 non-null  float64
 12  DataValueFootnoteSymbol    188019 non-null  object 
 13  DatavalueFootnote          18

In [7]:
df_uscronic.describe(include='all')

YearStart        YearEnd LocationAbbr    LocationDesc DataSource  \
count   403984.000000  403984.000000       403984          403984     403984   
unique            NaN            NaN           55              55         33   
top               NaN            NaN           NC  North Carolina      BRFSS   
freq              NaN            NaN         7779            7779     299921   
mean      2012.729851    2012.756698          NaN             NaN        NaN   
std          1.546388       1.515176          NaN             NaN        NaN   
min       2001.000000    2001.000000          NaN             NaN        NaN   
25%       2011.000000    2012.000000          NaN             NaN        NaN   
50%       2013.000000    2013.000000          NaN             NaN        NaN   
75%       2014.000000    2014.000000          NaN             NaN        NaN   
max       2016.000000    2016.000000          NaN             NaN        NaN   

           Topic                      Question Response DataValueUnit  \
count     403984                        403984    79323        374119   
unique        17                           202        1            14   
top     Diabetes  Mortality from heart failure                      %   
freq       67342                          6136    79323        280119   
mean         NaN                           NaN      NaN           NaN   
std          NaN                           NaN      NaN           NaN   
min          NaN                           NaN      NaN           NaN   
25%          NaN                           NaN      NaN           NaN   
50%          NaN                           NaN      NaN           NaN   
75%          NaN                           NaN      NaN           NaN   
max          NaN                           NaN      NaN           NaN   

           DataValueType  ...     LocationID  TopicID QuestionID  \
count             403984  ...  403984.000000   403984     403984   
unique                19  ...            NaN       17        201   
top     Crude Prevalence  ...            NaN      DIA     CVD1_4   
freq              150281  ...            NaN    67342       6136   
mean                 NaN  ...      31.003374      NaN        NaN   
std                  NaN  ...      17.700259      NaN        NaN   
min                  NaN  ...       1.000000      NaN        NaN   
25%                  NaN  ...      17.000000      NaN        NaN   
50%                  NaN  ...      30.000000      NaN        NaN   
75%                  NaN  ...      45.000000      NaN        NaN   
max                  NaN  ...      78.000000      NaN        NaN   

       DataValueTypeID  StratificationCategoryID1  StratificationID1  \
count           403984                     403984             403984   
unique              19                          3                 10   
top            CrdPrev                       RACE                OVR   
freq            150281                     246619              63215   
mean               NaN                        NaN                NaN   
std                NaN                        NaN                NaN   
min                NaN                        NaN                NaN   
25%                NaN                        NaN                NaN   
50%                NaN                        NaN                NaN   
75%                NaN                        NaN                NaN   
max                NaN                        NaN                NaN   

       StratificationCategoryID2 StratificationID2 StratificationCategoryID3  \
count                      79324             79324                     79323   
unique                         1                 1                         1   
top                                                                            
freq                       79324             79324                     79323   
mean                         NaN               NaN                       NaN   
std         

In [8]:
df_uscronic.shape

(403984, 34)

Este dataset incluye un total de 403,984 registros de muerte en Chile entre 2016 y 2021.

#Analisis resumido de las variables (Valores distintos)

In [9]:
for variable in df_uscronic.columns:
    print(variable)
    print(df_uscronic[variable].unique(),"\n")

YearStart
[2015 2013 2014 2012 2011 2010 2009 2016 2008 2007 2001] 

YearEnd
[2015 2013 2014 2012 2011 2010 2016 2008 2007 2001] 

LocationAbbr
['AK' 'AL' 'AR' 'AZ' 'CA' 'CO' 'CT' 'DC' 'FL' 'GA' 'GU' 'HI' 'IA' 'ID'
 'IL' 'IN' 'KS' 'KY' 'LA' 'MA' 'MD' 'ME' 'MI' 'MN' 'MO' 'MS' 'MT' 'NC'
 'ND' 'NE' 'NH' 'NJ' 'NM' 'NV' 'NY' 'OH' 'OK' 'OR' 'PA' 'PR' 'RI' 'SC'
 'SD' 'TX' 'US' 'UT' 'VA' 'VT' 'WA' 'WI' 'WV' 'WY' 'DE' 'TN' 'VI'] 

LocationDesc
['Alaska' 'Alabama' 'Arkansas' 'Arizona' 'California' 'Colorado'
 'Connecticut' 'District of Columbia' 'Florida' 'Georgia' 'Guam' 'Hawaii'
 'Iowa' 'Idaho' 'Illinois' 'Indiana' 'Kansas' 'Kentucky' 'Louisiana'
 'Massachusetts' 'Maryland' 'Maine' 'Michigan' 'Minnesota' 'Missouri'
 'Mississippi' 'Montana' 'North Carolina' 'North Dakota' 'Nebraska'
 'New Hampshire' 'New Jersey' 'New Mexico' 'Nevada' 'New York' 'Ohio'
 'Oklahoma' 'Oregon' 'Pennsylvania' 'Puerto Rico' 'Rhode Island'
 'South Carolina' 'South Dakota' 'Texas' 'United States' 'Utah' 'Virginia'
 'Ver

##Distribución de las variables en el dataset

In [10]:
for col in df_uscronic:
    print('* * * * * * * * * *',col,'* * * * * * * * * *')
    print(df_uscronic[col].value_counts(ascending=True))
    print(df_uscronic[col].value_counts(normalize=True))
    print('******************************************')

* * * * * * * * * * YearStart * * * * * * * * * *
2007       55
2001      104
2008      108
2016      109
2009     2771
2010    24965
2015    64046
2011    74593
2014    75524
2012    75796
2013    85913
Name: YearStart, dtype: int64
2013    0.212664
2012    0.187621
2014    0.186948
2011    0.184643
2015    0.158536
2010    0.061797
2009    0.006859
2016    0.000270
2008    0.000267
2001    0.000257
2007    0.000136
Name: YearStart, dtype: float64
******************************************
* * * * * * * * * * YearEnd * * * * * * * * * *
2007       55
2001      104
2008      108
2016      109
2010    24965
2015    64046
2011    74556
2014    75524
2012    76108
2013    88409
Name: YearEnd, dtype: int64
2013    0.218843
2012    0.188394
2014    0.186948
2011    0.184552
2015    0.158536
2010    0.061797
2016    0.000270
2008    0.000267
2001    0.000257
2007    0.000136
Name: YearEnd, dtype: float64
******************************************
* * * * * * * * * * LocationAbbr * * * * * * 

In [11]:
df_uscronic.shape   

(403984, 34)

##Contar valores NA por columna

In [12]:
print(df_uscronic.isna().sum())

YearStart                         0
YearEnd                           0
LocationAbbr                      0
LocationDesc                      0
DataSource                        0
Topic                             0
Question                          0
Response                     324661
DataValueUnit                 29865
DataValueType                     0
DataValue                    106167
DataValueAlt                 130318
DataValueFootnoteSymbol      215965
DatavalueFootnote            216131
LowConfidenceLimit           157165
HighConfidenceLimit          157165
StratificationCategory1           0
Stratification1                   0
StratificationCategory2      324661
Stratification2              324661
StratificationCategory3      324661
Stratification3              324661
GeoLocation                    2568
ResponseID                   324661
LocationID                        0
TopicID                           0
QuestionID                        0
DataValueTypeID             

##Eliminar columnas innecesarias

In [13]:
df_uscronic = df_uscronic.drop(columns=["DataSource",
"Response",
"DataValueFootnoteSymbol",
"DataValueAlt",
'StratificationCategory2',
"StratificationCategory3",
"Stratification2",
"Stratification3",
"ResponseID",
"QuestionID",
"TopicID",
"StratificationCategoryID1",
"StratificationCategoryID2",
"StratificationCategoryID3",
"StratificationID1",
"StratificationID2",
"StratificationID3"])
df_uscronic.shape   

(403984, 17)

In [14]:
df_uscronic.corr()

YearStart   YearEnd  LowConfidenceLimit  \
YearStart             1.000000  0.978520           -0.130898   
YearEnd               0.978520  1.000000           -0.128758   
LowConfidenceLimit   -0.130898 -0.128758            1.000000   
HighConfidenceLimit  -0.123367 -0.123784            0.977811   
LocationID            0.017846  0.016865           -0.007902   

                     HighConfidenceLimit  LocationID  
YearStart                      -0.123367    0.017846  
YearEnd                        -0.123784    0.016865  
LowConfidenceLimit              0.977811   -0.007902  
HighConfidenceLimit             1.000000   -0.009486  
LocationID                     -0.009486    1.000000

# Notas
Sin anotaciones por el momento